src/Compilers/Core/Portable/CommandLine/CommonCommandLineParser.cs

```diff
@@ -451,11 +457,25 @@ internal static string RemoveTrailingSpacesAndDots(string path)
                     continue;
                 }
 
-                if (string.Equals(arg, shared, StringComparison.OrdinalIgnoreCase))
+                if (arg.StartsWith(shared, StringComparison.OrdinalIgnoreCase))
                 {
+                    var hasValue = arg.Length > shared.Length && (arg[shared.Length] == ':' || arg[shared.Length] == '=');
```

Comment:
Don't want to take that approach because it establishes a policy of what values could be allowed in a session name.  Imagine for instance if I wanted to change my session name to be an URL or IP address / port combination.  That logic would disallow such specifications.

I prefer to treat the session name as an opaque identifier.